aws/resource_aws_db_instance.go

```diff
@@ -1451,6 +1502,17 @@ func resourceAwsDbInstanceUpdate(d *schema.ResourceData, meta interface{}) error
 		req.PreferredMaintenanceWindow = aws.String(d.Get("maintenance_window").(string))
 		requestUpdate = true
 	}
+	if d.HasChange("max_allocated_storage") {
+		d.SetPartial("max_allocated_storage")
+		mas := d.Get("max_allocated_storage").(int)
+
+		if mas == 0 {
```

Comment:
The correct way to disable mas is to set it to the current allocation? Wouldn’t that just set it to the current amount rather then disable ir🤔

Reply:
Ah let me drop a comment in there! This is designed to cover both these scenarios for disabling autoscaling:

* Setting the `max_allocated_storage` value equal to the `allocated_storage` value (the method documented in the AWS user guide)
* Removing the `max_allocated_storage` configuration altogether (the `0` check)

I think from an operator perspective the second scenario could be expected behavior and the API will return an error if you try to set it to `0` as its expecting the allocated storage value to disable autoscaling. 👍